### PR TITLE
feat(repo): creating pr test action

### PR DESCRIPTION
**Pull request by jasonmosley**
The purpose of this workflow is to build and run our tests when someone creates a PR. This will allow developers to catch issues sooner and proactively fix their PRs before they are reviewed.

### CODE REVIEW

**Comment by jasonmosley** [comment]
Do not merge until @xero reviews.

**Comment by jasonmosley** [comment]
This is working since it has found the failing test:
```
Summary of all failing tests
FAIL packages/plugin-leanplum/test/index.test.ts
  ● plugin-leanplum › ios
```

**Comment by jasonmosley** [comment]
So for the last step we are going to change the Flagship Code CLI to throw a process.exit(1) if it's being run in a CI and has a failure otherwise it will exit it 0. So as it stands now this PR is ready to be reviewed.

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  build:
+
+    runs-on: macos-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js
+      uses: actions/setup-node@v3
```
Comment (xero) [inline .github/workflows/pr-test.yml:15]: please update this version pins to be hashes

File: .github/workflows/pr-test.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    runs-on: macos-latest
+
+    steps:
+    - uses: actions/checkout@v3
```
Comment (xero) [inline .github/workflows/pr-test.yml:13]: please update to use pinned hashes `@v3` should be `@962678f22cdec338fba2a277a33cf81d5e28f7b9` i believe. but double check the upsteam

**Comment by xero** [comment]
checkout 
https://michaelheap.com/improve-your-github-actions-security/

**Review by xero — APPROVED** [review]
👁️ 👁️ 👍

**Review by crherman7 — APPROVED** [review]

**Pull request merged**